jira/client.py

```diff
@@ -3564,16 +3561,63 @@ def search_issues(
             use_post=use_post,
         )
 
-        if untranslate:
-            iss: Issue
-            for iss in issues:
-                for k, v in untranslate.items():
-                    if iss.raw:
-                        if k in iss.raw.get("fields", {}):
-                            iss.raw["fields"][v] = iss.raw["fields"][k]
-
+        self._untranslate_issue_fields_id_to_fields_name(issues, translation_map)
         return issues
 
+    def _translate_fields_name_to_id(
+        self, fields: list[str] | str
+    ) -> tuple[list[str], dict[str, str]]:
+        """Translate fields names to JIRA's REST API field id.
+
+        Args:
+            fields (list[str] | str): comma-separated string or list of issue fields to be translated
+
+        Returns:
+            tuple[list[str], dict[str, str]]: Returns translated fields and dictionary {field_id: field_name}
+            with translation needed to decode ids into names.
+            Translated fields variable is always List type
+        """
+        translated_fields: list[str]
+        if isinstance(fields, str):
+            translated_fields = fields.split(",")
+        elif isinstance(fields, list):
+            translated_fields = fields.copy()
+        else:
+            raise NotImplementedError(
+                "Translating fields for <%s> type is unsupported", type(fields)
+            )
+
+        translation_map = {}
+        if self._fields_cache:
+            for i, field_name in enumerate(translated_fields):
+                if field_name in self._fields_cache:
+                    translation_map[self._fields_cache[field_name]] = translated_fields[
+                        i
+                    ]
+                    translated_fields[i] = self._fields_cache[field_name]
+
+        return translated_fields, translation_map
+
+    @staticmethod
+    def _untranslate_issue_fields_id_to_fields_name(
+        issues: list[Issue], translation_map: dict[str, str]
+    ) -> None:
+        """Untranslate issues fields names from JIRA's REST API field id.
+
+        Args:
+            issues (list[Issue]): list of issues to be untranslated
+            translation_map (dict[str, str]): translation dict containing original field name and field id
+        """
+        if not translation_map:
+            return
```

Comment:
do we want to move this out of a staticmethod and instead have a cache of the the translation_map inside the class instance? ie similar to what is done with the `self._field_cache` ?